graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -183,19 +183,25 @@ struct SegmentEndFunc {
 
 c10::intrusive_ptr<sampling::FusedSampledSubgraph> SampleNeighbors(
     torch::Tensor indptr, torch::Tensor indices,
-    torch::optional<torch::Tensor> nodes, const std::vector<int64_t>& fanouts,
-    bool replace, bool layer, bool return_eids,
-    torch::optional<torch::Tensor> type_per_edge,
+    torch::optional<torch::Tensor> seeds,
+    torch::optional<std::vector<int64_t>> seed_offsets,
```

Comment:
could you add explanation here to elaborate how optimization is achieved under the help of `seed_offsets`?

Reply:
We compute the output of sample_neighbors + _convert_to_subgraph inside C++ in a fused manner. That is why it is faster. Will add the clarifications.

Reply:
Added.